backend.native/tests/runtime/workers/worker11.kt

```diff
@@ -40,9 +39,49 @@ fun initJobs(count: Int) = Array<Job?>(count) { i -> Job(i, i * 2, i)}
             consumed++
         }
     }
-    assertEquals(consumed, COUNT)
+    assertEquals(consumed, workers.size)
     workers.forEach {
         it.requestTermination().result
     }
     println("OK")
+}
+
+val COUNT = 2
+
+@SharedImmutable
+val counters = Array(COUNT) { AtomicInt(0) }.freeze()
+
+@Test fun runTest1() {
+  val workers = Array(COUNT) { Worker.start() }
+  // Ensure processQueue() can only be called on current Worker.
+  workers.forEach {
+      assertFailsWith<IllegalStateException> {
+          it.processQueue()
+      }
+  }
+  val futures = Array(workers.size) { workerIndex ->
+      workers[workerIndex].execute(TransferMode.SAFE, {
+          workerIndex
+      }) {
+          index ->
+          assertEquals(0, counters[index].value)
+          // Process following request.
+          while (!Worker.current!!.processQueue()) {}
```

Comment:
This test doesn't actually ensure that `processQueue` returns `false` if queue is empty.
The test will likely fail if `processQueue` returns `true` in this case, but it makes sense to avoid relying on non-deterministic effects.

Reply:
It's not what it tests, so for that there shall be a different test.